rclcpp/test/rclcpp/node_interfaces/test_node_topics.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <memory>
+#include <string>
+
+#include "rcl/node_options.h"
+#include "rclcpp/node.hpp"
+#include "rclcpp/node_interfaces/node_topics.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "test_msgs/msg/empty.hpp"
+
+static const rosidl_message_type_support_t empty_type_support =
+  *rosidl_typesupport_cpp::get_message_type_support_handle<test_msgs::msg::Empty>();
+
+static const rcl_publisher_options_t publisher_options =
+  rclcpp::PublisherOptionsWithAllocator<std::allocator<void>>().template
+  to_rcl_publisher_options<test_msgs::msg::Empty>(rclcpp::QoS(10));
+
+static const rcl_subscription_options_t subscription_options =
+  rclcpp::SubscriptionOptionsWithAllocator<std::allocator<void>>().template
+  to_rcl_subscription_options<test_msgs::msg::Empty>(rclcpp::QoS(10));
```

Comment:
@brawner hmm, consider turning these into functions from which you can retrieve these values. Using static, global storage for non-POD types is almost always dangerous.

Reply:
Switched these to functions.